src/doctypes/customizers.js

```diff
@@ -0,0 +1,176 @@
+import mergeWith from 'lodash/mergeWith'
+import mapValues from 'lodash/mapValues'
+
+// We trust the server by default
+const DEFAULT_POLICY = 'server'
+
+// When the value of an array is present on the server but not on
+// the user, by default we trust the user, otherwise a user could
+// not delete an item from an array
+const DEFAULT_DELETION_POLICY = 'user'
+
+const mergePolicies = {
+  user: (serverValue, userValue) => userValue,
+  server: serverValue => serverValue
+}
+
+const getCustomizerFromPolicy = policy => {
+  return mergePolicies[policy] || mergePolicies[DEFAULT_POLICY]
+}
+
+/**
+ * Can control what happens if case of deletion, should we trust the
+ * server or the user ?
+ */
+const deletionPolicies = {
+  user: group => group.user
```

Comment:
I don't understand the group concept here

Reply:
When merging two arrays, you first have to regroup values based on their ids. Here the group is an object containing values that have the same id, where group.user is the value coming from the user document, and group.server is the value coming from the server document.

Reply:
OK, thanks